opencog/nlp/learn/scm/link-pipeline.scm

```diff
@@ -461,36 +464,19 @@
  it parsed, and then updates the counts for the observed words and word
  pairs.
 "
-	; try-catch wrapper around the counters. Due to a buggy RelEx
-	; (see documentation for `word-inst-get-word`), the function
-	; `update-clique-pair-counts` might throw.  If it does throw,
-	; then avoid doing any counting at all for this sentence.
-	;
-	; Note: update-clique-pair-counts commented out. If you want this,
-	; then uncomment it, and adjust the length.
-	; Note: update-disjunct-counts commented out. It generates some
-	; data, but none of it will be interesting to most people.
-	(define (update-counts sent)
-		(catch 'wrong-type-arg
-			(lambda () (begin
-				; 6 == max distance between words to count.
-				; See docs above for explanation.
-				; (update-clique-pair-counts sent 6 #f)
-				(update-word-counts sent)
-				(update-lg-link-counts sent)
-				; If you uncomment this, be sure to also uncomment
-				; LgParseLink below, because LgParseMinimal is not enough.
-				; (update-disjunct-counts sent)
-			))
-			(lambda (key . args) #f)))
-
 	; Count the atoms in the sentence, according to the counting method
 	; passed as argument, then delete the sentence.
+
+	; Note: update-disjunct-counts commented out. It generates some
+	; data, but none of it will be interesting to most people.
 	(define (process-sent SENT cnt-mode win-size)
 		(update-word-counts SENT)
 		(if (equal? cnt-mode "any")
 			(update-lg-link-counts SENT)
 			(update-clique-pair-counts SENT win-size #f))
```

Comment:
Can you turn this into a case-statement? Currently, you wrote this so that if the mode is "any", then link-counting is used; otherwise, clique-pair counting is used as the default. It would be nicer to make this into an explicit case statement, with the two cases being "any" and "clique". 

Also -- weren't you planning on having other modes besides just these two? Or was this all that was envisioned?

Reply:
Yes, this was only an example of how I was planning to do the function wrappers for backwards compatibility. My next PR will implement the clique-mode counting which accounts for distance, and will be with a case-statement ;)

Reply:
OK